IDL and Sandbox preprocessor invocation doesn't have defines from GCC_PREPROCESSOR_DEFINITIONS

hortont424 opened the pull request:
#### 5ec2d0dd0c8f83a720b8d057bb60c6953eed85e8
<pre>
IDL and Sandbox preprocessor invocation doesn&apos;t have defines from GCC_PREPROCESSOR_DEFINITIONS
<a href="https://bugs.webkit.org/show_bug.cgi?id=249363">https://bugs.webkit.org/show_bug.cgi?id=249363</a>
rdar://103381924

Reviewed by Alexey Proskuryakov.

* Source/JavaScriptCore/DerivedSources.make:
* Source/WebCore/DerivedSources.make:
* Source/WebKit/DerivedSources.make:
Include defines from GCC_PREPROCESSOR_DEFINITIONS when calling out
to the preprocessor for IDL and Sandbox files. This can in some cases
cause mismatches between the features enabled in C++ code and in IDLs
or sandbox profiles.

Also set RELEASE_WITHOUT_OPTIMIZATIONS; otherwise in release builds,
GCC_PREPROCESSOR_DEFINITIONS gets us -NDEBUG, but not -O3, and Platform.h
complains about the unexpected mismatch.

Canonical link: <a href="https://commits.webkit.org/257915@main">https://commits.webkit.org/257915@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a2b88165d96cfab6f5e045cb00efdf131d0592dd

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/100353 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/9515 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/33416 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/109670 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/169888 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/104346 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/10427 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/52 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/92753 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/107548 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/106131 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/7891 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/91154 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/34541 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/89818 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/22546 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/77499 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/90901 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/3256 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/24063 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/86912 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/712 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/3249 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/173 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/29103 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/9370 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/43553 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/89795 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5436 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/5065 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/20075 "Passed tests") | 
| | | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov reviewed (approved):
Seems fine, assuming that it doesn't terribly conflict with Elliott's PR for feature flags.

hortont424 committed:
IDL and Sandbox preprocessor invocation doesn't have defines from GCC_PREPROCESSOR_DEFINITIONS
https://bugs.webkit.org/show_bug.cgi?id=249363
rdar://103381924

Reviewed by Alexey Proskuryakov.

* Source/JavaScriptCore/DerivedSources.make:
* Source/WebCore/DerivedSources.make:
* Source/WebKit/DerivedSources.make:
Include defines from GCC_PREPROCESSOR_DEFINITIONS when calling out
to the preprocessor for IDL and Sandbox files. This can in some cases
cause mismatches between the features enabled in C++ code and in IDLs
or sandbox profiles.

Also set RELEASE_WITHOUT_OPTIMIZATIONS; otherwise in release builds,
GCC_PREPROCESSOR_DEFINITIONS gets us -NDEBUG, but not -O3, and Platform.h
complains about the unexpected mismatch.

Canonical link: https://commits.webkit.org/257915@main
